feat(gcp): Adds GCP tags docs (#277)
meshcloud/meshcloud-docs: docs/meshstack.tag-schema.md

Operators can model this metadata as **tags** using a [tag schema](#tag-schemas). When a meshStack implementation has configured [tag schemas](#tag-schemas) it makes metadata tags available to orchestration and reports.

## Tag Schemas

[Configured tag-schemas](meshstack.tag-schema.md) describe the tags available on each meshModel entity and
their associated validation rules. These rules can describe for example if a tag is required or optional.

> meshStack currently supports tag schemas on [meshCustomers](meshcloud.customer.md) and [meshProjects](meshcloud.project.md) only.

> For each supported entity, meshStack can maintain two separate separate tag-schemas: **unrestricted** and **restricted** tags.

Reviewer: ```suggestion
For each supported entity, meshStack can maintain two separate tag-schemas: **unrestricted** and **restricted** tags.
```

Author: done

### Unrestricted Tags

Entity owners can edit tag values for **unrestricted tags** in self-service via meshPanel. For example, a user with a [Customer Admin](./meshcloud.customer.md#assign-meshcustomer-roles) role can edit tags values on the [meshCustomer](meshcloud.customer.md) as well as the [meshProjects](meshcloud.project.md) owned by that customer.

### Restricted Tags

> Only [Partners](./administration.index.md) can edit tag values for **restricted tags**. This is useful if an organization

Reviewer: Good point. But as far as I know partners can not change tag values through the meshPanel in general. Mostly the values can be changes by authoritative external systems.

Author: I just copy pasted the stuff :D I am not sure but I change the text accordingly.

needs to maintain metadata but doesn't want users to edit these values in self-service. For example, an organization can use restricted tags to implement a "segregation of duty" control to ensure that a partner
user confirmed a project's data classification.

## Configuring Tag Schemas

### Defining Tag Schemas

Operators define tag-schemas using JSON Schema. As described earlier on this page, meshStack uses metadata tags in a wide range of use
cases like providing parameters to Landing Zones. To ensure compatibility with different interfaces, tags must be simple
key-value pairs of strings only.

> The tag schema must therefore ensure all values are `string` values only. Operators that wish to represent numbers can

Reviewer: ```suggestion
The tag schema must therefore ensure all values are `string` values only. Operators who want to represent numbers can
```

Author: done
